Structure_Engine/Modify/Flip.cs

```diff
@@ -86,6 +144,123 @@ public static Stem Flip(this Stem stem)
 
             return flipped;
         }
+
+        /***************************************************/
+
+        private static ISectionProperty Flip(this ISectionProperty section)
+        {
+            if (section is IGeometricalSection geometricalSection)
+            {
+                ISectionProperty tempSection = null;
```

Comment:
Think we can add a pre-check here, and simply return the section if the profile is doubly symmetric. No need to re-create the same thing.

```suggestion
                if(geometricalSection.SectionProfile.ISymmetric() == Symmetry.DoublySymmetric)
                    return geometricalSection;

                ISectionProperty tempSection = null;
```